pkg/databind/internal/secrets/cyberarkapi.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package secrets
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/pkg/databind/pkg/data"
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+	"os"
+)
+
+type CyberArkAPI struct {
+	HTTP *http
+}
+
+type cyberArkAPIGatherer struct {
+	cfg *CyberArkAPI
+}
+
+// CyberArkAPIGatherer instantiates a CyberArkAPI variable gatherer from the given configuration.
+// The result is a map with a single "password" key value pair
+func CyberArkAPIGatherer(cyberArkAPI *CyberArkAPI) func() (interface{}, error) {
+	g := cyberArkAPIGatherer{cfg: cyberArkAPI}
+	return func() (interface{}, error) {
+		dt, err := g.get()
+		if err != nil {
+			return "", err
+		}
+		return dt, err
+	}
+}
+
+func (g *cyberArkAPIGatherer) get() (data.InterfaceMap, error) {
+	secret := g.cfg
+	dt, err := httpRequest(secret.HTTP, "GET", nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve cyberarkapi secret from http server: %s", err)
+	}
+
+	smap := data.InterfaceMap{}
+	if err := json.Unmarshal(dt, &smap); err != nil {
+		return nil, fmt.Errorf("unable to decode cyberarkapi secret: %s", err)
+	}
+	if p, ok := smap["Content"]; ok {
+		if u, ok := smap["UserName"]; ok {
+			result := data.InterfaceMap{}
+			result["password"] = p.(string)
+			result["user"] = u.(string)
+			fmt.Fprintf(os.Stderr, "get: returning %v\n\n", result)
```

Comment:
Why do we need to print to Stderr retrieved data?

Reply:
If you want to log something use the the logger. The logger can be found [here](https://github.com/newrelic/infrastructure-agent/blob/master/pkg/databind/internal/secrets/secretys.go#L10) and an example of it being used can be found [here](https://github.com/newrelic/infrastructure-agent/blob/master/pkg/databind/internal/secrets/kms.go#L105). Also, don't log passwords, secrets or anything else that is sensitive. If you need to do so then obfuscate using [this](https://github.com/newrelic/infrastructure-agent/blob/1d1b5c0634863852966fa541fc0b4224892075b3/pkg/helpers/helpers.go#L348) but better just not to print it.

Reply:
Hmm... that shouldn't have been there, I changed it to the logger... I'll take a look.

Reply:
Fixed